Slider/Assets/Scripts/Misc/SimpleInteractableCutscene.cs

```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    public bool Interact()
+    {
+        TrySkipCurrentDialogue();
+
+        return true;
+    }
+
+    private void TrySkipCurrentDialogue()
+    {
+        if (currentlyTalkingCharacter != null)
+        {
+            if (currentlyTyping)
+            {
+                currentlyTalkingCharacter.SkipText();
+            }
+            else if (currentlyWaitingAfterTyped)
+            {
+                skipWaitingAfterTyped = true;
+            }
+        }
+    }
+
+    private void StartCutScene()
+    {
+        foreach (NPC character in cutsceneCharacters)
+        {
+            if (character.IsDialogueBoxActive())
+            {
+                character.DeactivateDialogueBox();
+            }
+        }
+        StartCoroutine(StartCutSceneCoroutine());
+    }
+
+    private IEnumerator StartCutSceneCoroutine()
+    {
+        cutsceneStarted = true;
+
+        if (cutsceneStartedFlag != null)
+        {
+            SaveSystem.Current.SetBool(cutsceneStartedFlag, true);
+        }
+
+        EnableAllNormalCharacterDialogueTriggers(false);
+
+        yield return CutScene();
+
+        if (playerInTrigger)
+        {
+            AllowPlayerInteraction(false);
+        }
+
+        cutsceneFinished = true;
+
+        if (cutsceneFinishedFlag != null)
+        {
+            SaveSystem.Current.SetBool(cutsceneFinishedFlag, true);
+        }
+        //we want a bit of a gap after cutscene is done before you can start talking to them like normal, otherwise it's weird
+        yield return new WaitForSeconds(2f);
+
+        EnableAllNormalCharacterDialogueTriggers(true);
+    }
+
+    private void EnableAllNormalCharacterDialogueTriggers(bool enable)
+    {
+        foreach (NPC character in cutsceneCharacters)
+        {
+            EnableNormalCharacterDialogueTrigger(character, enable);
+        }
+    }
+
+    private void EnableNormalCharacterDialogueTrigger(NPC character, bool enable)
+    {
+        OnTriggerEnter triggerEnterScript = character.GetComponent<OnTriggerEnter>();
+        if (triggerEnterScript != null)
+        {
+            triggerEnterScript.SetOnPlayerEnterActive(enable);
+        }
+    }
+
+    protected virtual IEnumerator CutScene()
+    {
+        yield return null;
+    }
+   
+    protected IEnumerator SayNextDialogue(NPC character, float timeWaitAfterFinishedTyping = DEFAULT_TIME_BETWEEN_DIALOGUE_LINES)
+    {
+        character.TypeCurrentDialogue();
+        currentlyTalkingCharacter = character;
+        currentlyTyping = true;
+
+        yield return new WaitWhile(() => character.IsTypingDialogue());
+
+        currentlyTyping = false;
+        currentlyWaitingAfterTyped = true;
+
+        float startTime = Time.time;
+        float currentTime = Time.time;
+        while ((currentTime - startTime < timeWaitAfterFinishedTyping) && !skipWaitingAfterTyped)
+        {
+            yield return new WaitForFixedUpdate();
```

Comment:
do `yield return null;` here instead to wait for a normal frame (fixed update is a physics frame)